src/SFML.Window/WindowBase.cs

```diff
@@ -0,0 +1,797 @@
+        /// </summary>
+        ////////////////////////////////////////////////////////////
+        public void DispatchEvents()
+        {
+            Event e;
+            while (PollEvent(out e))
+            {
+                CallEventHandler(e);
+            }
+        }
+
+        ////////////////////////////////////////////////////////////
+        /// <summary>
+        /// Request the current window to be made the active
+        /// foreground window
+        /// </summary>
+        ////////////////////////////////////////////////////////////
+        public virtual void RequestFocus()
+        {
+            sfWindowBase_requestFocus(CPointer);
+        }
+
+        ////////////////////////////////////////////////////////////
+        /// <summary>
+        /// Check whether the window has the input focus
+        /// </summary>
+        /// <returns>True if the window has focus, false otherwise</returns>
+        ////////////////////////////////////////////////////////////
+        public virtual bool HasFocus()
+        {
+            return sfWindowBase_hasFocus(CPointer);
+        }
+
+        ////////////////////////////////////////////////////////////
+        /// <summary>
+        /// Create a Vulkan rendering surface
+        /// </summary>
+        /// <param name="vkInstance">Vulkan instance</param>
+        /// <param name="vkSurface">Created surface</param>
+        /// <param name="vkAllocator">Allocator to use</param>
+        /// <returns>True if surface creation was successful, false otherwise</returns>
+        ////////////////////////////////////////////////////////////
+        public virtual bool CreateVulkanSurface(IntPtr vkInstance, out IntPtr vkSurface, IntPtr vkAllocator)
+        {
+            return sfWindowBase_createVulkanSurface(CPointer, vkInstance, out vkSurface, vkAllocator);
+        }
+
+        ////////////////////////////////////////////////////////////
+        /// <summary>
+        /// Provide a string describing the object
+        /// </summary>
+        /// <returns>String description of the object</returns>
+        ////////////////////////////////////////////////////////////
+        public override string ToString()
+        {
+            return "[WindowBase]" +
+                   " Size(" + Size + ")" +
+                   " Position(" + Position + ")";
+        }
+
+        ////////////////////////////////////////////////////////////
+        /// <summary>
+        /// Constructor for derived classes
+        /// </summary>
+        /// <param name="cPointer">Pointer to the internal object in the C API</param>
+        /// <param name="dummy">Internal hack :)</param>
+        ////////////////////////////////////////////////////////////
+        protected WindowBase(IntPtr cPointer, int dummy) :
+            base(cPointer)
+        {
+            // TODO : find a cleaner way of separating this constructor from WindowBase(IntPtr handle)
```

Comment:
Can you describe what the issue currently is?

Reply:
This is pretty much the same issue replicated from SFML.Window to SFML.WindowBase. The constructor for initializing a window from an OS handle has the same signature as the one to create a Window object from a CSFML pointer (i.e. both use IntPtr), so the extra dummy parameter is used to differentiate them. #240 might be an option to fix it, though.